proteus/NonlinearSolvers.py

```diff
@@ -617,6 +618,102 @@ def solve(self,u,r=None,b=None,par_u=None,par_r=None):
+                 maxLSits = 100):
+        Newton.__init__(self,
+                linearSolver,
+                F,J,du,par_du,
+                rtol_r,
+                atol_r,
+                rtol_du,
+                atol_du,
+                maxIts,
+                norm,
+                convergenceTest,
+                computeRates,
+                printInfo,
+                fullNewton,
+                directSolver,
+                EWtol,
+                maxLSits)
+        self.DB = 11
+        U = np.loadtxt('SVD_basis')
+        self.U = U[:,0:self.DB]
+        self.U_transpose = self.U.conj().T
+        self.pod_J = np.zeros((self.DB,self.DB),'d')
+        self.pod_linearSolver = LU(self.pod_J)
+        self.J_rowptr,self.J_colind,self.J_nzval = self.J.getCSRrepresentation()
+        self.pod_du = np.zeros(self.DB)
+    def norm(self,u):
+        return self.norm_function(u)
+    def solve(self,u,r=None,b=None,par_u=None,par_r=None):
+        """
+        Solve F(u) = b
+
+        b -- right hand side
+        u -- solution
+        r -- F(u) - b
+        """
+        pod_u = np.dot(self.U_transpose,u)
+        u[:] = np.dot(self.U,pod_u)
+        r=self.solveInitialize(u,r,b)
+        pod_r = np.dot(self.U_transpose,r)
+        self.norm_r0 = self.norm(pod_r)
+        self.norm_r_hist = []
+        self.norm_du_hist = []
+        self.gammaK_max=0.0
+        self.linearSolverFailed = False
+        while (not self.converged(pod_r) and
+               not self.failed()):
+            log("   Newton it %d norm(r) = %12.5e  \t\t norm(r)/(rtol*norm(r0)+atol) = %g test=%s"
+                % (self.its-1,self.norm_r,(self.norm_r/(self.rtol_r*self.norm_r0+self.atol_r)),self.convergenceTest),level=1)
+            if self.updateJacobian or self.fullNewton:
+                self.updateJacobian = False
+                self.F.getJacobian(self.J)
+                self.pod_J[:] = 0.0
+                for i in range(self.DB):
+                    for j in range(self.DB):
+                        for k in range(self.F.dim):
+                            for m in range(self.J_rowptr[k],self.J_rowptr[k+1]):
+                                self.pod_J[i,j] += self.U_transpose[i,k]*self.J_nzval[m]*self.U[self.J_colind[m],j]
+                #self.linearSolver.prepare(b=r)
+                self.pod_linearSolver.prepare(b=pod_r)
+            self.du[:]=0.0
+            self.pod_du[:]=0.0
+            if not self.linearSolverFailed:
+                #self.linearSolver.solve(u=self.du,b=r,par_u=self.par_du,par_b=par_r)
+                #self.linearSolverFailed = self.linearSolver.failed()
+                self.pod_linearSolver.solve(u=self.pod_du,b=pod_r)
+                self.linearSolverFailed = self.pod_linearSolver.failed() 
+            #pod_u-=np.dot(self.U_transpose,self.du)
+            pod_u-=self.pod_du
+            u[:] = np.dot(self.U,pod_u)
+            self.computeResidual(u,r,b)
+            pod_r[:] = np.dot(self.U_transpose,r)
```

Comment:
Initially I just replace this line with the `PF` projection applied to `r[rho]`.